src/main/java/org/opensearch/ad/caching/CacheBuffer.java

```diff
@@ -240,16 +241,26 @@ public boolean canRemove() {
         ModelState<EntityModel> valueRemoved = items.remove(keyToRemove);
 
         if (valueRemoved != null) {
-            // if we releasing a shared cache item, release memory as well.
-            if (!reserved) {
-                memoryTracker.releaseMemory(memoryConsumptionPerEntity, false, Origin.MULTI_ENTITY_DETECTOR);
+            if (reserved) {
+                // release in reserved memory
+                memoryTracker.releaseMemory(memoryConsumptionPerEntity, true, Origin.HC_DETECTOR);
+            } else {
+                // release in shared memory
+                memoryTracker.releaseMemory(memoryConsumptionPerEntity, false, Origin.HC_DETECTOR);
             }
-            checkpointDao.write(valueRemoved, keyToRemove);
-        }
 
-        EntityModel modelRemoved = valueRemoved.getModel();
-        if (modelRemoved != null) {
-            modelRemoved.clear();
+            EntityModel modelRemoved = valueRemoved.getModel();
+            if (modelRemoved != null) {
+                if (modelRemoved.getRcf() == null || modelRemoved.getThreshold() == null) {
+                    // only have samples. If we don't save, we throw the new samples and might
+                    // never be able to initialize the model
+                    checkpointWriteQueue.write(valueRemoved, true, SegmentPriority.MEDIUM);
+                } else {
+                    checkpointWriteQueue.write(valueRemoved, false, SegmentPriority.MEDIUM);
+                }
```

Comment:
[minor] Similar to above, could simply to:
```
// not sure what to name this variable since I can't find the class CheckpointWriteQueue and read the function signature
boolean isNullModel = modelRemoved.getRcf() == null || modelRemoved.getThreshold() == null;
checkpointWriteQueue.write(valueRemoved, isNullModel, SegmentPriority.MEDIUM);
```

Reply:
yeah, changed using your version